packages/salesforcedx-utils-vscode/src/helpers/utils.ts

```diff
@@ -14,11 +16,40 @@ export function isNullOrUndefined(object: any): object is null | undefined {
 }
 
 export function extractJsonObject(str: string): any {
-
-  const jsonString = str.substring(
-    str.indexOf('{'),
-    str.lastIndexOf('}') + 1
-  );
+  const jsonString = str.substring(str.indexOf('{'), str.lastIndexOf('}') + 1);
 
   return JSON.parse(jsonString);
 }
+
+// There's a bug in VS Code where, after a file has been renamed,
+// the URI that VS Code passes to the command is stale and is the
+// original URI.  See https://github.com/microsoft/vscode/issues/152993.
+//
+// To get around this, fs.realpathSync.native() is called to get the
+// URI with the actual file name.
+export function flushFilePath(filePath: string): string {
+  if (filePath === '') {
+    return filePath;
+  }
+
+  let nativePath = fs.realpathSync.native(filePath);
+  if (/^win32/.test(process.platform)) {
+    // The file path on Windows is in the form of "c:\Users\User Name\foo.cls".
+    // When called, fs.realpathSync.native() is returning the file path back as
+    // "C:\Users\User Name\foo.cls", and the capitalization of the drive letter
+    // causes problems further down stream.  To fix this, we'll use the path
+    // returned from fs.realpathSync.native() and then capitalize the first
+    // character.
+    nativePath = nativePath.charAt(0).toLowerCase() + nativePath.slice(1);
```

Comment:
👍🏽

Reply:
stupid questions b/c windows.  The default is C:\ but can a user have a drive named a lower case letter?  So would d:\ be valid vs D:\?  There may be a corner case here were a drive is actually a lower case letter (a quick google told me this does happen in some instances, like when using a usb drive).  We probably need a follow on ticket here to address the corner case and having the ability to ask the filesystem the valid path.

Reply:
From what I can tell, what's displayed is always uppercase (`D:\`) but what the APIs/SDK uses is always lowercase (`d:\`).  I tested this on a VM running Windows, but it's VERY slow, so I'd like a more thorough test on a real Windows machine b/f we merge.

Reply:
(also, if I remember correctly, Windows is case-insensitive, so `d:\` is the same as `D:\`.  When one performs a `FileExists("c:\\foo\\bar.txt")`, it's the same as `FileExists("C:\\FOO\\BAR.TXT")`

Reply:
I'm full of dumb questions.  With the case insensitivity of windows why are we required to make the above change?  It's been too long I don't remember what the 'down stream' changes are that result in the issue.

Reply:
That's a really good question, and I wondered about that as well, but unfortunately, I don't know why the casing is failing.  When I was debugging this, I noticed that the path was "c:\bla\bla\bla.cls" but fs.realpathSync.native() was returning "C:\bla\bla\bla.cls", and when I changed the case for the first char (the drive letter), our actions (deploy, delete, etc...) worked again.